compiler/src/iree/compiler/Modules/IO/Parameters/IR/IOParametersOps.td

```diff
@@ -0,0 +1,237 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_DIALECT_MODULES_IO_PARAMETERS_OPS
+#define IREE_DIALECT_MODULES_IO_PARAMETERS_OPS
+
+include "iree/compiler/Dialect/HAL/IR/HALBase.td"
+include "iree/compiler/Dialect/Util/IR/UtilAttrs.td"
+include "iree/compiler/Dialect/Util/IR/UtilInterfaces.td"
+include "iree/compiler/Modules/IO/Parameters/IR/IOParametersBase.td"
+include "mlir/IR/OpAsmInterface.td"
+include "mlir/IR/SymbolInterfaces.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+class IOParameters_PureOp<string mnemonic, list<Trait> traits = []> :
+    IOParameters_Op<mnemonic, !listconcat(traits, [Pure])>;
+
+//===----------------------------------------------------------------------===//
+// Parameter I/O
+//===----------------------------------------------------------------------===//
+
+def OpGroupParameterOps : OpDocGroup {
+  let summary = "Parameter I/O ops";
+  let description = "Ops parameter I/O.";
+}
+
+let opDocGroup = OpGroupParameterOps in {
+
+def IOParameters_LoadOp : IOParameters_Op<"load", [
+  Util_SizeAwareOp,
+]> {
+  let summary = [{reads a parameter from a parameter scope}];
+  let description = [{
+    Asynchronously reads a parameter from an external parameter provider and
+    returns the resulting buffer. Depending on the parameter and buffer types
```

Comment:
Just a thought while still early in the review: can we see parameter loads in Tracy? I'm thinking of the network panel in Chrome Dev Tools as the sort of UI I might want (though gather/scatter could complicate that?)

Reply:
Yep! Everything is instrumented so we can see individual parameters getting read/written with their keys associated with the zones.